invoice/src/main/java/org/killbill/billing/invoice/dao/InvoiceDaoHelper.java

```diff
@@ -378,6 +419,26 @@ public boolean apply(final Tag input) {
         });
     }
 
+    private void setParentInvoice(final InvoiceModelDao invoice, final List<Tag> invoicesTags, final EntitySqlDaoWrapperFactory entitySqlDaoWrapperFactory, final InternalTenantContext childContext) {
+        final InvoiceParentChildrenSqlDao invoiceParentChildrenSqlDao = entitySqlDaoWrapperFactory.become(InvoiceParentChildrenSqlDao.class);
+        final List<InvoiceParentChildModelDao> mappings = invoiceParentChildrenSqlDao.getParentChildMappingsByChildInvoiceIds(ImmutableList.of(invoice.getId().toString()), childContext);
+        if (mappings.isEmpty()) {
+            return;
+        }
+
+        Preconditions.checkState(mappings.size() == 1, String.format("Expected only one parent mapping for invoice", invoice.getId()));
```

Comment:
⚠️ Missing format parameter in the `String.format` (also, `Preconditions.checkState` can call `String.format` for you - check the varargs signature one).

Reply:
Ack will fix.